skytemple/module/symbols/widget/main.py

```diff
@@ -0,0 +1,308 @@
+                    .set_rw_data(rw_symbol, binary_id, protocol)
+                    .build()
+                )
+                symbol_entry.register(self.entry_list, self.symbols_treestore)
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_binary_list_changed(self, binary_list_element: Gtk.ComboBoxText):
+        self.symbols_treefilter.refilter()
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_search_input_changed(self, search_entry_element: Gtk.SearchEntry):
+        self.symbols_treefilter.refilter()
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_cr_value_completion_editing_started(self, widget: Gtk.CellRenderer, editable: Gtk.CellEditable, path: str):
+        if isinstance(editable, Gtk.Entry):
+            store_entry = self.symbols_treefilter[path]
+            model = store_entry[12]
+            self.symbols_completion.set_model(model)
+            editable.set_completion(self.symbols_completion)
+        else:
+            raise NotImplementedError()
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_cr_text_value_changed(self, widget: Gtk.CellRendererText, path: str, new_value: str, *args):
+        self.on_value_changed(path, new_value)
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_cr_bool_value_changed(self, widget: Gtk.CellRendererToggle, path: str, *args):
+        # The internal state of the widget is updated after the signal handler is run, so to get the new value, we
+        # need to invert the current state.
+        new_value = not widget.get_active()
+        self.on_value_changed(path, "1" if new_value else "0")
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_cr_combo_value_changed(self, widget: Gtk.CellRendererCombo, path: str, new_iter: Gtk.TreeIter, *args):
+        # Retrieve the numeric ID of the new value
+
+        # WORKAROUND: For some reason, model-linked properties in the widget are None sometimes when this method
+        # is called, even if they are set in the model (in particular, this happens only if the entry is part of
+        # a struct).
+        # To fix this, we pull the model directly from the TreeFilter.
+        model = self.symbols_treefilter[path][12]
+        new_value = model[new_iter][0]
+        self.on_value_changed(path, str(new_value), new_iter)
+
+    # noinspection PyUnusedLocal
+    @Gtk.Template.Callback()
+    def on_cr_completion_value_changed(self, widget: Gtk.CellRendererText, path: str, new_value: str, *args):
+        match = COMPLETION_ENTRY_REGEX.search(new_value)
+        if match:
+            _id = match.group(1)
+            # Convert to int and then back to string to remove leading zeroes
+            new_value_final = str(int(_id))
+            self.on_value_changed(path, new_value_final)
+
+    def on_value_changed(self, path: str, new_value: str, model_iter: Optional[Gtk.TreeIter] = None):
```

Comment:
Where exactly should the asterisk appear? I don't think I can access the individual cells to change the text.
Bolding should be doable though. Do I just bold the entire row?

Reply:
Sure.
As for the asterisks, there are a few options, potentially the eaiest is to add a new cell renderer to the first column that renders a column in the store that either contains "" or "*", this is similiar to how it works in the main item store.

Reply:
Uh... how do I know when the ROM has been saved so I can restore the normal text display?

Reply:
Ah, right. I don't remember if there is a hook for that, maybe not. But tbh I think it's also fine if you don't reset it while the symbol editor is still open for now.
I guess another issue is also that you don't know what changed when you re-open. But that is also fine.
I guess this way it's more like a "this is what you changed since you last opened this". In that case just using bold is probably better, no asterisk, since the asterisk is assosiated with the ROM itself not being saved, that might confuse people.

Reply:
Alright then, just bold for now.